### DISCUSSION TITLE
OSError: [Errno 1] EPERM

### DISCUSSION THREAD

**Discussion by FlydGitHub** [micropython/micropython#11060] [RP2040 / Pico] [answered]
Hi Can anyone help with a wifi error?

OSError: [Errno 1] EPERM

The context is line 15:  wlan.connect(ssid, password)

I have tried everything to fix this without success.

My platform is the 2040 Challenger NB - 8MB,

Thanks

**Comment by davefes** [comment]
EPERM looks like a Windows error not Micropython. Try:
```py
try:
    wlan.connect(ssid, password)
except OSError as error:
    print(f'error is {error}')
```

**Comment by FlydGitHub** [comment]
Thanks Dave,

Now I get:
>>> %Run -c $EDITOR_CONTENT
error is [Errno 1] EPERM
waiting for connection...
waiting for connection...
----dito------
waiting for connection...
Traceback (most recent call last):
  File "<stdin>", line 50, in <module>
RuntimeError: network connection failed
>>>

**Comment by FlydGitHub** [comment]
Line 50:

  raise RuntimeError('network connection failed')

**Comment by davefes** [comment]
Have never seen a RuntumeError() but then again I haven't used the Pico.  Seems different to the many problems I have had connecting to WiFi on the ESP32.

Are you running an IDE and also what editor are you using?

**Comment by FlydGitHub** [comment]
1. Thonny - current version
2. Mac
3. The Challenger has 8MB and wifi is provided by an onboard ESP8285 chip for wifi

**Reply by davefes** [reply]
Which Micropython image are you running?

Show us the full wifi connect method.

These devices will start in the AP mode.

For a bullet-proof connect script see here:
https://github.com/glenn20/micropython-espnow-utils/tree/main/src
and wifi.py

**Comment by FlydGitHub** [comment]
1. Image = March 13 2023 - current version.
2. #https://www.tomshardware.com/how-to/connect-raspberry-pi-pico-w-to-the-internet

import network
import secrets
import time
import urequests
wlan = network.WLAN(network.STA_IF)
wlan.active(True)
wlan.connect(secrets.SSID, secrets.PASSWORD)
print(wlan.isconnected())
astronauts = urequests.get("http://api.open-notify.org/astros.json").json()
number = astronauts['number']
for i in range(number):
    print(astronauts['people'][i]['name'])

3. I see a consistent issue with MP, I install a module, and it still companies it's not found, eg  in this case 1st line network

**Reply by robert-hh** [reply]
Ŷour board seems not directly to be supported by MicroPython. And if you use the standard RP2040 firmware, then there is not network module. RP2040 boards with WiFi support are e.g. the Pico W board or the Arduino  Nano RP2040 connect. Does Challenger supply a MicroPython compatible WiFi network module for your board?

**Comment by FlydGitHub** [comment]
The claim is the the Challenger supports both MicroPython and CiriutPython

**Reply by robert-hh** [reply]
I looked at their web site, and it's a typical marketing claim. Yes, they support MicroPython and CircuitPython. But they do not tell to which extend. No word about WiFi and MicroPython or CircuitPython.

**Comment by FlydGitHub** [comment, marked as answer]
Thank you Robert, I very much appreciate you checking out there website.

**Reply by robert-hh** [reply]
I was interested which WiFi module they use. It's based on a ESP8285 chip with the standard  espressif AT interface. They provide support for it in the Arduino libraries.
